### PR TITLE
Gerbil

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Many notable changes to Gerbil, that itself depends on notable changes to Gambit, have happened in the two months or so since I last upstreamed my changes to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/gerbil/default.nix [hunk 1]
```diff
@@ -7,11 +7,11 @@
 stdenv.mkDerivation rec {
   name    = "gerbil-${version}";
 
-  version = "0.12-DEV-777-gd855915";
+  version = "0.12-DEV-836-gcde6802";
```
Comment (Mic92) [inline pkgs/development/compilers/gerbil/default.nix]: same here.

File: pkgs/development/compilers/gambit/default.nix [hunk 2]
```diff
@@ -4,13 +4,13 @@
 
 stdenv.mkDerivation rec {
   name    = "gambit-${version}";
-  version = "4.8.8-300db59";
+  version = "4.8.8-415-g29ed48bb";
```
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/compilers/gambit/default.nix]: Maybe `415` is already enough?

**Comment by grahamc** [comment]
@GrahamcOfBorg eval

**Comment by fare** [comment]
Indeed the 415 and 836 are the number of commits and make it easy to order the version numbers. These versions are taken from git describe --tags (stripping the leading v from the tag, so nix recognizes it's version numbers indeed).

I can switch to date if you prefer, but the above is what Gerbil natively uses, and that Gambit may use in the future.

File: pkgs/development/compilers/gambit/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (fare) [reply pkgs/development/compilers/gambit/default.nix]: I believe 415 is OK.

**Comment by 7c6f434c** [comment]
I think if upstream explicitly references commit counts, it is an aceptable style of patchlevel.

**Comment by Mic92** [comment]
@7c6f434c yes, but version still contains a commit hash in pr you have just merged, not just the patchlevel.

**Comment by 7c6f434c** [comment]
Well, hash after the patchlevel should not be able to break version
comparison.

**Pull request merged**